Bluetooth: controller: Update supported commands for ISO

wopu-ot opened the pull request:
Set bits for ISO-related supported commands depending on selected
features and adjust compilation guards for the respective handler
functions.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

wopu-ot committed:
Bluetooth: controller: Update supported commands for ISO

Set bits for ISO-related supported commands depending on selected
features and adjust compilation guards for the respective handler
functions.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

wopu-ot committed:
Bluetooth: controller: Fix use of handle field in HCI command handling

Fix use of handle field in HCI command handling for some ISO-related
commands. The response buffer may reuse the command buffer. Therefore,
the handle field must not be copied directly from the command to the
response.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

cvinayak commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -1939,7 +1968,23 @@ static void le_remove_cig(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_iso_transmit_test *cmd = (void *)buf->data;
+	struct bt_hci_rp_le_iso_transmit_test *rp;
+	uint8_t status;
+	uint16_t handle;
```
```suggestion
	uint16_t handle;
	uint8_t status;
```

cvinayak commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -1963,6 +2008,46 @@ static void le_read_iso_tx_sync(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_iso_receive_test *cmd = (void *)buf->data;
+	struct bt_hci_rp_le_iso_receive_test *rp;
+	uint8_t status;
+	uint16_t handle;
```
```suggestion
	uint16_t handle;
	uint8_t status;
```

cvinayak commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -1963,6 +2008,46 @@ static void le_read_iso_tx_sync(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_read_test_counters *cmd = (void *)buf->data;
+	struct bt_hci_rp_le_read_test_counters *rp;
+	uint8_t status;
+	uint16_t handle;
+	uint32_t received_cnt;
+	uint32_t missed_cnt;
+	uint32_t failed_cnt;
```
```suggestion
	uint32_t received_cnt;
	uint32_t missed_cnt;
	uint32_t failed_cnt;
	uint16_t handle;
	uint8_t status;
```

cvinayak reviewed (commented):
Reviewed the bitmap, LGTM.

Minor nitpick on the reverse Christmas tree declarations.

wopu-ot committed:
Bluetooth: controller: Reorder variable declarations in ISO HCI commands

Reorder variable declarations in HCI command handlers of ISO-related
commands to conform to reverse Christmas tree style.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

wopu-ot commented:
Put on DNM because bit for LE ISO Transmit Test is missing and LE Read ISO Link Quality should be optional. I will update accordingly.

Thalley commented:
> Put on DNM because bit for LE ISO Transmit Test is missing and LE Read ISO Link Quality should be optional. I will update accordingly.

@wopu-ot How about `LE Set CIG Parameters Test`?

wopu-ot committed:
Bluetooth: controller: Introduce Kconfig for Read ISO Link Quality

Introduce Kconfig for Read ISO Link Quality command. Support for the
command is optional according to the specification.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

wopu-ot commented:
> @wopu-ot How about `LE Set CIG Parameters Test`?
It's in line 737/738.

Thalley commented:
> > @wopu-ot How about `LE Set CIG Parameters Test`?
> 
> It's in line 737/738.

Sorry, I wasn't clear: Shouldn't `LE Set CIG Parameters Test` be optional as well?

wopu-ot commented:
> > > @wopu-ot How about `LE Set CIG Parameters Test`?
> > 
> > 
> > It's in line 737/738.
> 
> Sorry, I wasn't clear: Shouldn't `LE Set CIG Parameters Test` be optional as well?

That command is not an optional command, it is mandatory if the CIS central role is supported.

BT Core Spec Version 5.3, Vol. 4, Part E, Table 3.1:
> LE Set CIG Parameters Test command ... C39
> ...
> C39: Mandatory if LE Feature (Connected Isochronous Stream - Central) is supported,
otherwise excluded.

Thalley commented:
> > > > @wopu-ot How about `LE Set CIG Parameters Test`?
> > > 
> > > 
> > > It's in line 737/738.
> > 
> > 
> > Sorry, I wasn't clear: Shouldn't `LE Set CIG Parameters Test` be optional as well?
> 
> That command is not an optional command, it is mandatory if the CIS central role is supported.
> 
> BT Core Spec Version 5.3, Vol. 4, Part E, Table 3.1:
> 
> > LE Set CIG Parameters Test command ... C39
> > ...
> > C39: Mandatory if LE Feature (Connected Isochronous Stream - Central) is supported,
> > otherwise excluded.

I see, thanks. To me it's curious that you can't disable support for this command in production builds, but I suppose there's some reason for that.
